firefox-ios/Client/TabManagement/Tab.swift

```diff
@@ -1193,9 +1252,37 @@ class TabWebView: WKWebView, MenuHelperWebViewInterface, ThemeApplicable {
     /// Updates the `background-color` of the webview to match
     /// the theme if the webview is showing "about:blank" (nil).
     func applyTheme(theme: Theme) {
+        documentLoadingView?.applyTheme(theme: theme)
         if url == nil {
             let backgroundColor = theme.colors.layer1.hexString
             evaluateJavascriptInDefaultContentWorld("document.documentElement.style.backgroundColor = '\(backgroundColor)';")
         }
     }
+
+    // MARK: - Document loading handling
+
+    func showDocumentLoadingView() {
```

Comment:
This is currently living under the `TabWebView` which isn't an issue at firsthand, but if we consider the `WebEngine` then this has some repercussion. `TabWebView` will be a `DefaultWKEngineWebView` in the future. The `WebEngine` shouldn't handle any UI (such as a `TemporaryDocumentLoadingView`) since we don't have access to theming/strings/etc in that package (and we shouldn't). This mean the `WebEngine` will ask the Client to show this `TemporaryDocumentLoadingView` when needed. If we could have a similar pattern already in production this would help by a lot for the project. Ideally we would also create another ticket to port this feature inside the WebEngine.

Reply:
So after discussion, we'll create a task under the webview epic so we adjust the solution there so the WebEngine doesn't create the UI itself. We won't adjust the production code to work like that since it won't easily support it like the WebEngine will.